mkt/prices/models.py

```diff
@@ -347,7 +347,13 @@ def create_addon_purchase(sender, instance, **kw):
                      c=instance))
         return
 
-    if instance.type == mkt.CONTRIB_PURCHASE:
+    print
+    print dir(instance.price_tier)
+    print instance.price_tier.price
+    print instance.price_tier.name
+
+    if instance.type == mkt.CONTRIB_PURCHASE \
+            and not instance.price_tier.name == '0':
```

Comment:
actually, I think we rely on Contribution to store the record of a user purchasing an in-app product. Thus, you could skip AddonPurchase for all in-app purchases (maybe). But that's a separate bug.